synapse/tests/test_lib_lmdbslab.py

```diff
@@ -47,22 +47,67 @@ class LmdbSlabTest(s_t_utils.SynTest):
             self.eq(b'hehe', slab.get(b'\x00\x01', db=foo))
 
             items = list(slab.scanByPref(b'\x00', db=foo))
-            self.eq(items, ((b'\x00\x01', b'hehe'), (b'\x00\x02', b'haha'), (b'\x00\x03', b'hoho')))
+            self.eq(items, ((b'\x00\x01', b'hehe'), (b'\x00\x02', b'haha')))
 
-            items = list(slab.scanByRange(b'\x00\x02', b'\x00\x03', db=foo))
-            self.eq(items, ((b'\x00\x02', b'haha'), (b'\x00\x03', b'hoho')))
+            items = list(slab.scanByRange(b'\x00\x02', b'\x01\x03', db=foo))
+            self.eq(items, ((b'\x00\x02', b'haha'), (b'\x01\x03', b'hoho')))
 
             items = list(slab.scanByDups(b'\x00\x02', db=bar))
             self.eq(items, ((b'\x00\x02', b'haha'), (b'\x00\x02', b'visi'), (b'\x00\x02', b'zomg')))
 
+            # backwards scan tests
+
+            items = list(slab.scanByPrefBack(b'\x00', db=foo))
+            self.eq(items, ((b'\x00\x02', b'haha'), (b'\x00\x01', b'hehe')))
+
+            items = list(slab.scanByPrefBack(b'\x01', db=foo))
+            self.eq(items, ((b'\x01\x03', b'hoho'),))
+
+            items = list(slab.scanByRangeBack(b'\x00\x03', db=foo))
+            self.eq(items, ((b'\x00\x02', b'haha'), (b'\x00\x01', b'hehe')))
+
+            items = list(slab.scanByRangeBack(b'\x00\x03', b'\x00\x02', db=foo))
+            self.eq(items, ((b'\x00\x02', b'haha'), ))
+
+            items = list(slab.scanByRangeBack(b'\x01\x03', b'\x00\x02', db=foo))
+            self.eq(items, ((b'\x01\x03', b'hoho'), (b'\x00\x02', b'haha')))
+
+            items = list(slab.scanByRangeBack(b'\x01\x05', b'\x00\x02', db=foo))
+            self.eq(items, ((b'\x01\x03', b'hoho'), (b'\x00\x02', b'haha')))
+
+            items = list(slab.scanByDupsBack(b'\x00\x02', db=bar))
+            self.eq(items, ((b'\x00\x02', b'zomg'), (b'\x00\x02', b'visi'), (b'\x00\x02', b'haha')))
+
+            items = list(slab.scanByDupsBack(b'\x00\x04', db=bar))
+            self.eq(items, ())
+
+            items = list(slab.scanByFullBack(db=foo))
+            self.eq(items, ((b'\x01\x03', b'hoho'), (b'\x00\x02', b'haha'), (b'\x00\x01', b'hehe')))
+
```

Comment:
I'd like to see a test with existing keys of '\xff' and '\xffff` and scanByPrefBack. I'm pretty sure it works correctly, but it would still be good to test.

Reply:
Test added.